src/MCPClient/lib/clientScripts/antivirus.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+# This file is part of Archivematica.
+#
+# Copyright 2010-2017 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+import argparse
+import multiprocessing
+import os
+import uuid
+
+import django
+
+django.setup()
+from clamav_client.scanner import get_scanner
+from custom_handlers import get_script_logger
+from databaseFunctions import insertIntoEvents
+from django.conf import settings as mcpclient_settings
+from django.core.exceptions import ValidationError
+from django.db import transaction
+from main.models import Event
+from main.models import File
+
+logger = get_script_logger("archivematica.mcp.client.clamscan")
+
+
+def concurrent_instances():
+    return multiprocessing.cpu_count()
+
+
+def file_already_scanned(file_uuid):
+    return (
+        file_uuid != "None"
+        and Event.objects.filter(
+            file_uuid_id=file_uuid, event_type="virus check"
+        ).exists()
+    )
+
+
+def queue_event(file_uuid, date, scanner, passed, queue):
+    if passed is None or file_uuid == "None":
+        return
+
+    event_detail = ""
+    if scanner is not None:
+        info = scanner.info()  # This is cached.
+        event_detail = f'program="{info.name}"; version="{info.version}"; virusDefinitions="{info.virus_definitions}"'
+
+    outcome = "Pass" if passed else "Fail"
+    logger.info("Recording new event for file %s (outcome: %s)", file_uuid, outcome)
+
+    queue.append(
+        {
+            "fileUUID": file_uuid,
+            "eventIdentifierUUID": str(uuid.uuid4()),
+            "eventType": "virus check",
+            "eventDateTime": date,
+            "eventDetail": event_detail,
+            "eventOutcome": outcome,
+        }
+    )
+
+
+def get_parser():
+    """Return a ``Namespace`` with the parsed arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("file_uuid", metavar="fileUUID")
+    parser.add_argument("path", metavar="PATH", help="File or directory location")
+    parser.add_argument("date", metavar="DATE")
+    parser.add_argument(
+        "task_uuid", metavar="taskUUID", help="Currently unused, feel free to ignore."
+    )
+    return parser
```

Comment:
Should we remove this helper function if we're not using it?

Reply:
Yup! Thanks.